registry/registry.go

```diff
@@ -105,7 +109,8 @@ func DockerHeaders(userAgent string, metaHeaders http.Header) []transport.Reques
 	modifiers := []transport.RequestModifier{}
 	if userAgent != "" {
 		modifiers = append(modifiers, transport.NewHeaderRequestModifier(http.Header{
-			"User-Agent": []string{userAgent},
+			"User-Agent":   []string{userAgent},
+			"X-Session-ID": []string{SessionID},
```

Comment:
Registry would consume either `X-Session-ID` or `Docker-Session-ID` to be compatible with systems that may inject `X-Session-ID` already.